backend/capellacollab/projects/capellamodels/modelsources/t4c/routes.py

```diff
@@ -40,45 +49,48 @@
     response_model=list[ResponseT4CModel],
 )
 def list_t4c_models(
-    db: Session = Depends(database.get_db),
-    token: JWTBearer = Depends(JWTBearer()),
-    model=Depends(get_existing_capella_model),
+    project: DatabaseProject = Depends(get_existing_project),
+    model: DatabaseCapellaModel = Depends(get_existing_capella_model),
     repository: t.Optional[DatabaseT4CRepository] = Depends(
-        load_instance_repository
+        get_existing_instance_repository
     ),
+    db: Session = Depends(database.get_db),
+    token=Depends(JWTBearer()),
 ):
-    verify_admin(token, db)
     if not repository:
-        return crud.get_all_t4c_models(db, model)
+        verify_project_role(project.name, token, db)
+        return model.t4c_models
     return T4CRepositoryWithModels.from_orm(repository).models
```

Comment:
```suggestion
    return repository.models
```
There are a few other places in the t4c routes where `from_orm` is called implicitly, all of which can be removed.